### PR TITLE
fix(adapter): harmonize context and set process.env

### LINKED ISSUE #90: Provide easy check for production/simulator environment
```javascript
    return Boolean(process.env.__OW_ACTIVATION_ID) || Boolean(process.env.AWS_LAMBDA_FUNCTION_NAME);
```

> I'd prefer to have something in the wrapper, but it's good for now.

_Originally posted by @trieloff in https://github.com/adobe/helix-pipeline/pull/947#discussion_r566130708_

### LINKED ISSUE #118: Harmonize context.func
the values in context.func are different for each target:

| name | example | desc |
|-|-|-|
| `name` | `dispatch` | stemmed name of the action |
| `version` | `4.3.1` | (dotted) version of the action |
| `package` | `helix-services` | package of the action |
| `app` | `helix` | app id / name of the action |
| `fqn` | `/helix/helix-services/dispatch@4.3.1` | fully qualified name, env dependent |
| `fqn` | `arn:aws:lambda:us-east-1:118435662149:function:helix-services--dispatch:4_4_13` | |

### PR DESCRIPTION
**Pull request by tripodsan**
fixes #118
fixes #90
closes #190 

@trieloff I'm not sure about the google and azure mapping for `context.func` (see #118)

I added tests for google. maybe you can add one for azure, otherwise I do it tomorrow.

### CODE REVIEW

File: src/template/index.js [hunk 1]
```diff
@@ -76,18 +68,21 @@ async function azure(context, req) {
       body,
     });
 
+    const [,,,, packageName, name, version, ...suffix] = req.url.split('/');
```
Comment (tripodsan) [inline src/template/index.js:71]: I'm not sure about this. maybe provide an example as comment?

File: src/template/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ */
+export declare interface FunctionInfo {
+  /**
+   * Function name. Stemmed name of the function.
+   * @example 'dispatch'
+   */
+  name:string;
+
+  /**
+   * Version in dotted format.
```
Comment (trieloff) [inline src/template/index.d.ts:51]: What about @ci versions?

File: src/template/index.js [hunk 3]
```diff
@@ -201,9 +199,11 @@ async function openwhiskAdapter(params = {}) {
         region: process.env.__OW_REGION,
       },
       func: {
-        name: names.join('/'),
-        version: process.env.__OW_ACTION_VERSION,
-        app: namespace,
+        name,
+        version,
+        package: packageName,
+        app: process.env.__OW_NAMESPACE,
```
Comment (tripodsan) [inline src/template/index.js:205]: btw: there is a difference between the namespace in the `__OW_ACTION_NAME` and the `__OW_NAMESPACE`.
the first is the one where the function is deployed, and the later is the one where it is executed.
I think the latter is more important. the first can still be retrieved using the `fqn`

File: src/template/index.js [hunk 4]
```diff
@@ -267,6 +268,7 @@ async function google(req, res) {
 
     const [subdomain] = req.headers.host.split('.');
     const [country, region, ...servicename] = subdomain.split('-');
+    const [packageName, name] = process.env.K_SERVICE.split('--');
```
Comment (tripodsan) [inline src/template/index.js:271]: not sure about this. @trieloff  can you give an example of the `K_SERVICE` ?

**Review by trieloff — APPROVED** [review]
Looks good. As for Azure, as long as the integration tests pass, we are ggod. I will add more Azure tests as we deploy real workloads there.

File: src/template/index.js (hunk 4 above)
Comment (trieloff) [reply src/template/index.js:271]: `simple-package--simple-name`

File: src/template/index.d.ts (hunk 2 above)
Comment (tripodsan) [reply src/template/index.d.ts:51]: that's ok. I added this to highlight that it's not the the same as the `_` format we use in the AWS aliases.

**Pull request merged**